team/bundles/org.eclipse.core.net/src/org/eclipse/core/internal/net/ProxyManager.java

```diff
@@ -66,11 +66,13 @@ public class ProxyManager implements IProxyService, IPreferenceChangeListener {
 	private ProxyManager() {
 		try {
 			String os = System.getProperty("osgi.os"); //$NON-NLS-1$
-			nativeProxyProvider = switch (os) {
-			case Constants.OS_LINUX -> new ProxyProviderLinux();
-			case Constants.OS_WIN32 -> new ProxyProviderWin32();
-			default -> null;
-			};
+			if (os != null) {
+				nativeProxyProvider = switch (os) {
+				case Constants.OS_LINUX -> new ProxyProviderLinux();
+				case Constants.OS_WIN32 -> new ProxyProviderWin32();
+				default -> null;
```

Comment:
@laeubi you could also just write ...`case null, default -> null;`. The switch can deal with null, if it's declared in the cases.

Reply:
I'm not sure if its not only the case on Java 21,... but even `System.getProperty("osgi.os", "")` would be possible... so many options :-D

Reply:
right, I sometimes forget that not everyone is on 21 yet. :)